src/link/link_info/info_port.rs

```diff
@@ -107,23 +116,20 @@ impl InfoPortData {
         payload: &[u8],
         kind: InfoPortKind,
     ) -> Result<InfoPortData, DecodeError> {
-        Ok(match kind {
-            InfoPortKind::Bond => {
-                let mut v = Vec::new();
-                for nla in NlasIterator::new(payload) {
-                    let nla = &nla.context(format!(
-                        "failed to parse IFLA_INFO_PORT_DATA \
-                    (IFLA_INFO_PORT_KIND is '{kind}')"
-                    ))?;
-                    let parsed = InfoBondPort::parse(nla).context(format!(
-                        "failed to parse IFLA_INFO_PORT_DATA \
-                    (IFLA_INFO_PORT_KIND is '{kind}')"
-                    ))?;
-                    v.push(parsed);
-                }
-                InfoPortData::BondPort(v)
-            }
-            InfoPortKind::Other(_) => InfoPortData::Other(payload.to_vec()),
-        })
+        let port_data = match kind {
+            InfoPortKind::Bond => NlasIterator::new(payload)
+                .map(|nla| nla.and_then(|nla| InfoBondPort::parse(&nla)))
+                .collect::<Result<Vec<_>, _>>()
```

Comment:
You discarded the error message using this hack-like codes. Please restored to original code layout.

Reply:
I think it's not discarded. This `collect` is converting `Vec<Result<T, E>>` to `Result<Vec<T>, E>`. If any of the previous `nla` or `parse` is an `Err`, this will return that `Err`.
Below port_data.context(...)? will add the context and throw the error.

Reply:
What do you mean by "hack-like"? Yes, the code is less imperative but there is no hack.. everything seems legit..

Reply:
My bad. I read the code wrong. Yes, the error message is preserved.